src/main/java/com/laytonsmith/abstraction/bukkit/events/BukkitPlayerEvents.java

```diff
@@ -969,4 +1016,544 @@ public void setCancelled(boolean bln) {
+
+		PlayerResourcePackStatusEvent prpse;
+
+		public BukkitMCPlayerResourcepackStatusEvent(PlayerResourcePackStatusEvent e) {
+			this.prpse = e;
+		}
+
+		@Override
+		public MCPlayer getPlayer() {
+			return new BukkitMCPlayer(prpse.getPlayer());
+		}
+
+		@Override
+		public CString getStatus() {
+			return new CString(prpse.getStatus().name(), Target.UNKNOWN);
+		}
+
+		@Override
+		public Object _GetObject() {
+			return prpse;
+		}
+	}
+
+	@abstraction(type = Implementation.Type.BUKKIT)
+	public static class BukkitMCPlayerRiptideEvent implements MCPlayerRiptideEvent {
+
+		PlayerRiptideEvent pre;
+
+		public BukkitMCPlayerRiptideEvent(PlayerRiptideEvent e) {
+			this.pre = e;
+		}
+
+		@Override
+		public MCItemStack getItem() {
+			return new BukkitMCItemStack(pre.getItem());
+		}
+
+		@Override
+		public MCPlayer getPlayer() {
+			return new BukkitMCPlayer(pre.getPlayer());
+		}
+
+		@Override
+		public Object _GetObject() {
+			return pre;
+		}
+	}
+
+	@abstraction(type = Implementation.Type.BUKKIT)
+	public static class BukkitMCPlayerStatisticIncrementEvent implements MCPlayerStatisticIncrementEvent {
+
+		PlayerStatisticIncrementEvent psie;
+
+		public BukkitMCPlayerStatisticIncrementEvent(PlayerStatisticIncrementEvent e) {
+			this.psie = e;
+		}
+
+		@Override
+		public EntityType getEntityType() {
+			return psie.getEntityType();
+		}
+
+		@Override
+		public Object getMaterial() {
+			return (psie.getMaterial() == null ? CNull.NULL : new BukkitMCMaterial(psie.getMaterial()));
+		}
+
+		@Override
+		public CInt getPreviousValue() {
+			return new CInt(psie.getPreviousValue(), Target.UNKNOWN);
+		}
+
+		@Override
+		public CInt getNewValue() {
+			return new CInt(psie.getNewValue(), Target.UNKNOWN);
+		}
+
+		@Override
+		public Statistic getStatistic() {
+			return psie.getStatistic();
+		}
+
+		@Override
+		public MCPlayer getPlayer() {
+			return new BukkitMCPlayer(psie.getPlayer());
+		}
+
+		@Override
+		public boolean isCancelled() {
+			return psie.isCancelled();
+		}
+
+		@Override
+		public void setCancelled(boolean cancel) {
+			psie.setCancelled(cancel);
+		}
+
+		@Override
+		public Object _GetObject() {
+			return psie;
+		}
+	}
+
+	@abstraction(type = Implementation.Type.BUKKIT)
+	public static class BukkitMCPlayerVelocityEvent implements MCPlayerVelocityEvent {
+
+		PlayerVelocityEvent pve;
+
+		public BukkitMCPlayerVelocityEvent(PlayerVelocityEvent e) {
+			this.pve = e;
+		}
+
+		@Override
+		public Vector getVelocity() {
+			return pve.getVelocity();
+		}
+
+		@Override
+		public MCPlayer getPlayer() {
+			return new BukkitMCPlayer(pve.getPlayer());
+		}
+
+		@Override
+		public boolean isCancelled() {
+			return pve.isCancelled();
+		}
+
+		@Override
+		public void setCancelled(boolean cancel) {
+			pve.setCancelled(cancel);
+		}
+
+		@Override
+		public void setVelocity(Vector velocity) {
+			pve.setVelocity(velocity);
```

Comment:
Part 6: (continued from part 5) ... and then by doing the conversion to Bukkit Vector here instead of in the function.